src/compiler/emitter.ts

```diff
@@ -2349,7 +2484,14 @@ namespace ts {
             emitList(parentNode, typeArguments, ListFormat.TypeArguments);
         }
 
-        function emitTypeParameters(parentNode: Node, typeParameters: NodeArray<TypeParameterDeclaration>) {
+        function emitTypeParameters(parentNode: SignatureDeclaration | InterfaceDeclaration | TypeAliasDeclaration | ClassDeclaration | ClassExpression, typeParameters: NodeArray<TypeParameterDeclaration>) {
+            if (parentNode.kind !== SyntaxKind.InterfaceDeclaration &&

```

Comment:
Should we throw an exception if the type arguments are defined for these syntax kinds?

Reply:
They won't be because the member doesn't exist on them. Typechecker's got our back here; the reason the if statement is here is to safely narrow to `SignatureDeclaration`, which are the only types which could have them member defined on them.